---
title: "Assignment2"
author: "Student Name: Yue Zhou"
output: 
  html_document: 
    toc: yes
---

## Indroduction
### Data background
Centrosomal protein 120 (CEP120) is a 120kD centrosome protein that plays an important role in centrosome replication. Overexpression of CEP120 can lead to centrosome amplification, which is closely associated with tumorigenesis and development. The authors want to know whether the overexpression of CEP120 will promot centrosome amplification in gastric cancer (GC).  

### Data source
The expression data was downloaded from GEO with ID GSE217567.This is a gastric cancer data.There are 6 samples(3 case and 3 control).CEP120.1~3 are cases and Vector1~3 are controls.The materials are cell line.Three case cell line were CEP120-overexpressed and the other three control cell line were not treated in anyway. 

### Basic Stats

```{r message=FALSE,warning=FALSE}
library(edgeR)
readscount<-read.table('GSE217567_processed_count.txt',header=TRUE)
readscount<-readscount[!duplicated(readscount$Gene),]
rownames(readscount)<-readscount$Gene
readscount<-readscount[,-1]
group <- rep(c('control', 'case'), each = 3)
dgelist <- DGEList(counts = readscount, group = group)
keep <- rowSums(cpm(dgelist) > 1 ) >= 2
#The barplot shows the gene number before and after filtering the low expression genes.
barplot(height = cbind(rawGeneCount=nrow(readscount),filterLowExpression=length(keep[keep==TRUE])),
        main = 'Fig.1 Gene numbers before and after filter')
dgelist <- dgelist[keep, , keep.lib.sizes = FALSE]
dgelist_norm <- calcNormFactors(dgelist, method = 'TMM')
plotMDS(dgelist_norm, col = rep(c('red', 'blue'), each = 3), dim = c(1, 2),
        main='Fig.2 MDS Plot')
```

## Differential Gene Expression
```{r message=FALSE,warning=FALSE}
design <- model.matrix(~group)    
dge <- estimateDisp(dgelist_norm, design, robust = TRUE)
fit <- glmFit(dge, design, robust = TRUE)    
lrt <- glmLRT(fit) 
write.csv(topTags(lrt, n = nrow(dgelist$counts)),file = 'rawResults.csv')
rawResult<-read.csv('rawResults.csv',header = TRUE,row.names = 1)
```
#### 1.Calculate p-values for each of the genes in your expression set. How many genes were significantly differentially expressed? What thresholds did you use and why?
14 genes were significantly differentially expressed.I used pvalue<0.05 and abs(logFC)>1 as the threshold.Because it is the most direct standard.
```{r message=FALSE,warning=FALSE}
nrow(rawResult[which(rawResult$PValue<0.05 & abs(rawResult$logFC)>1),])
```
#### 2.Multiple hypothesis testing - correct your p-values using a multiple hypothesis correction method. Which method did you use? And Why? How many genes passed correction?
I used FDR as a multiple hypothesis correction method.And only 3 genes passed correction.
```{r message=FALSE,warning=FALSE}
nrow(rawResult[which(rawResult$FDR<0.05 & abs(rawResult$logFC)>1),])
```
#### 3.Show the amount of differentially expressed genes using an MA Plot or a Volcano plot. Highlight genes of interest.
```{r message=FALSE,warning=FALSE}
library(ggplot2)
library(ggpubr)
library(ggrepel)
rawResult$logFDR<-(-log10(rawResult$FDR))
rawResult$logPvalue<-(-log10(rawResult$PValue))
rawResult$Change<-ifelse(rawResult$PValue < 0.05 & abs(rawResult$logFC) > 1, 
                          ifelse(rawResult$logFC> 1 ,'Up','Down'),
                          'Stable')
rawResult$Label<-ifelse(rawResult$PValue < 0.05 & abs(rawResult$logFC) > 1,as.character(rownames(rawResult)), "")
ggscatter(rawResult,x = "logFC",y = "logPvalue",
          color = "Change",
          palette = c("#546de5", "#d2dae2","#ff4757"),
          repel = T,
          ylab = "-log10(PValue)",
          size = 2) + 
  geom_hline(yintercept = 1.3,linetype = "dashed")+
  geom_vline(xintercept = c(-1,1),linetype = "dashed")+
  geom_label_repel(data = rawResult, aes(x = rawResult$logFC, 
                                     y = -log10(rawResult$PValue), 
                                     label = Label),
                 size = 3, box.padding = unit(0.5, "lines"),
                 point.padding = unit(0.8, "lines"), 
                 segment.color = "black", 
                 show.legend = FALSE)+
  ggtitle('Fig.3 Vocalno Plot of DEG')
```

#### 4.Visualize your top hits using a heatmap.Do you conditions cluster together?Explain why or why not.
Yes.My conditions cluster together.Beause the CEP120 does influence the expression of gastric cancer cell.
```{r message=FALSE,warning=FALSE}
library(pheatmap)
DEG_genes <- rawResult[rawResult$PValue<0.05 & abs(rawResult$logFC)>1,]
DEG_gene_expr <- readscount[rownames(DEG_genes),]
pheatmap(DEG_gene_expr,
         color = colorRampPalette(c("blue","white","red"))(100),
         scale = "row", 
         border_color = NA, 
         fontsize = 10, 
         show_rownames = F,
         legend = TRUE,
         main = 'Fig.4 Heatmap of DEGs')

```

## Thresholded over-representation analysis
#### 1.Which method did you choose and why?
I chose clusterProfiler to do GSEA analysis.Because it is convenient and easy to use.

#### 2.What annotation data did you use and why? What version of the annotation are you using?
I use c5.all.v2023.1.Hs.entrez.gmt as my annotation file.Because it is the GO genese sets and it is the most use for function enrichment analysis.I use the C5 2023 version.

#### 3.How many genesets were returned with what thresholds?
17 genesets were returned with pvalue<0.05.

#### 4.Run the analysis using the up-regulated set of genes, and the down-regulated set of genes separately. How do these results compare to using the whole list ?
There are 17 significantly results with all differentially expressed genes.There are 6 significantly results with all up regulated genes.There are 31 significantly results with all down regulated genes.  
The pathways are also different.I have listed the significantly results in Figure 5~7.
```{r fig.height=15, fig.width=10, message=FALSE, warning=FALSE}
library(clusterProfiler)
library(org.Hs.eg.db)
library(ggridges)
library(ggplot2)
library(fgsea)
rawResult<-read.csv('rawResults.csv',header = TRUE,row.names = 1)
DEG_genes <- rawResult[rawResult$PValue<0.05,]
DEG_gene_expr <- readscount[rownames(DEG_genes),]
DEG_genes$SYMBOL<-rownames(DEG_genes)
gene_names<-bitr(rownames(DEG_genes), fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
DEG_genes<-merge(DEG_genes,gene_names,by='SYMBOL')
rankfile<-DEG_genes[,c(7,2)]
rankfile<-rankfile[order(rankfile$logFC,decreasing = TRUE),]
gmt<-read.gmt('c5.all.v2023.1.Hs.entrez.gmt')
#1.all genesets 
genelist<-rankfile[,2]
names(genelist)<-as.character(rankfile[,1])
#gsea_all<-GSEA(genelist,TERM2GENE = gmt,pvalueCutoff=1)
#save(gsea_all,file = 'gsea_all.RData')
#number of genesets with all differential expressed genes
load('gsea_all.RData')
nrow(gsea_all@result[which(gsea_all@result$pvalue<0.05),])
ridgeplot(gsea_all,nrow(gsea_all@result[which(gsea_all@result$pvalue<0.05),]),fill = 'pvalue')+ggtitle('Fig.5 GSEA result with all genes')
#2.up genesets
genelist_up<-genelist[genelist>0]
#gsea_up<-GSEA(genelist_up,TERM2GENE = gmt,pvalueCutoff=1)
#number of genesets with up genes
load('gsea_up.RData')
nrow(gsea_up@result[which(gsea_up@result$pvalue<0.05),])
ridgeplot(gsea_up,nrow(gsea_up@result[which(gsea_up@result$pvalue<0.05),]),fill = 'pvalue')+ggtitle('Fig.6 GSEA result with up regulated genes')
#3.down gensets
genelist_down<-genelist[genelist<0]
#gsea_down<-GSEA(genelist_down,TERM2GENE = gmt,pvalueCutoff=1)
#number of genesets with down genes
load('gsea_down.RData')
nrow(gsea_down@result[which(gsea_down@result$pvalue<0.05),])
ridgeplot(gsea_down,nrow(gsea_down@result[which(gsea_down@result$pvalue<0.05),]),fill = 'pvalue')+ggtitle('Fig.7 GSEA result with down regulated genes')
```

## References
[1]Robinson MD, McCarthy DJ and Smyth GK (2010). edgeR: a Bioconductor package for differential expression analysis of digital gene expression data. Bioinformatics 26, 139-140.  
[2]H. Wickham. ggplot2: Elegant Graphics for Data Analysis. Springer-Verlag New York, 2016.  
[3]Alboukadel Kassambara (2020). ggpubr: 'ggplot2' Based Publication Ready Plots. R package version 0.4.0.  
[4]Kamil Slowikowski (2021). ggrepel: Automatically Position Non-Overlapping Text Labels with 'ggplot2'. R package version 0.9.1.  
[5]Raivo Kolde (2019). pheatmap: Pretty Heatmaps. R package version 1.0.12.   
[6]T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141.  
[7]Marc Carlson (2021). org.Hs.eg.db: Genome wide annotation for Human. R package version 3.13.0.  
[8]Claus O. Wilke (2022). ggridges: Ridgeline Plots in 'ggplot2'. R package version 0.5.4.  
[9]G. Korotkevich, V. Sukhov, A. Sergushichev. Fast gene set enrichment analysis. bioRxiv (2019), doi:10.1101/060012.





















